lib/WasmReader/WasmBinaryReader.h

```diff
@@ -94,6 +94,7 @@ namespace Wasm
             virtual WasmOp ReadFromBlock() override;
             virtual WasmOp ReadFromCall() override;
             virtual WasmOp ReadExpr() override;
+            virtual void Unread() override { m_pc--; m_funcState.count--; }
```

Comment:
It looks like `Unread` is used only to get the last read opcode. Why not just have a variable that keeps track of last opcode read ?
Also, `Unread` is confusing since you don't really "unread" it, you just go back so you can read it again.